buildpack/dockerfile.go

```diff
@@ -23,3 +30,171 @@ type ExtendArg struct {
 	Name  string `toml:"name"`
 	Value string `toml:"value"`
 }
+
+type InstructionCallback func(verb string, arg string, lineno int, state map[string]string) (string, error)
+
+func walkDockerInstructions(filename string, handleInstruction InstructionCallback) (string, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	var inContinueBlock bool = false
+	var checkLine bool = true
+	var lineNo = 0
+	var state = make(map[string]string)
+	for scanner.Scan() {
+		line := scanner.Text()
+		lineNo++
+
+		//skip comments (docker permits whitespace before #)
+		trimmedLine := strings.TrimSpace(line)
+		if strings.HasPrefix("#", trimmedLine) {
+			continue
+		}
+
+		//does this line start a continuation block?
+		if !inContinueBlock {
+			checkLine = true
+		} else {
+			checkLine = false
+		}
+		if strings.HasSuffix("\\", line) {
```

Comment:
Should we also trim spaces here?

Reply:
I don't think so. 
The continuation char must be the last char on the line, if there are spaces after it, it stops being a valid continuation char.. 
So here we have to test against `line` rather than `trimmedLine` because we care about the '\' chars original placement.

Reply:
Resolved, moved to mobi/buildkit parsing.